search-sentiments-map/public/scripts/current-data-script.js

```diff
@@ -51,3 +66,25 @@ function setNewTrend(trend) {
       });
 }
 
+/** 
+ * Retrieves relevant data for new trend and reconstructs map with new data.
+ * @param {string} topic New topic to get data for.
+ * @param {Array} countries Countries to get data for.
+ */
+function setUserSearchTopic(topic, countries) {
+  fetch('/trends').then(globalTrends => globalTrends.json()).then(trends => {
+    topTrends = trends;
+    setTopTrends(); 
+  });
```

Comment:
Since you are repeating this in two functions, can you make it a separate function? Also, consider calling it from setTopTrends in trends.js.